dojo/tools/horusec/parser.py

```diff
@@ -62,7 +62,7 @@ def _get_finding(self, data, date):
                 data["vulnerabilities"]["details"].split("\n")[-1],
                 "**Code:**",
                 f"```{data['vulnerabilities']['language']}",
-                data["vulnerabilities"]["code"].replace("```", "``````"),
+                data["vulnerabilities"]["code"].replace("```", "``````").replace("\x00", ""),
```

Comment:
Here's what this Finding looks like today if you just remove these null characters:

![Screenshot 2024-04-17 at 12 57 17](https://github.com/DefectDojo/django-DefectDojo/assets/1749665/61019ad6-2950-4883-bbe8-8d4aece8110d)

A couple thoughts:
* Ideally, we shouldn't throw away data reported by the tool like this - nulls might be meaningful sometimes (though they're not in this case as this seems like an obvious false positive, and they wouldn't be displayed anyway)
* Displaying random binary characters isn't the ideal behavior here, and this probably isn't the only tool where binary data could appear like this

I'm okay with approving this one for now since it fixes an open issue, but it seems like there might be a better approach to handling binary data like this more generally, such as using the `\uXXXX` formatting like the JSON example. Unfortunately, as this case demonstrates, it's not always obvious that a given field in a tool's report will *sometimes* contain binary data.

Reply:
Good hint, thank you @cneill